tests/JsonReaderTests.cs

```diff
@@ -657,6 +657,27 @@ public void Object_General_With_Valid_Input()
         TestValidInput(KeyIntMapReader, json, new() { ["foo"] = 123, ["bar"] = 456, ["baz"] = 789 });
     }
 
+    static readonly IJsonReader<Dictionary<string, Dictionary<string, string>>> NestedObjectReader =
+        JsonReader.Object(JsonReader.Object(JsonReader.String(),
+                                            ps => ps.ToDictionary(e => e.Key, e => e.Value)),
+                          ps => ps.ToDictionary(e => e.Key, e => e.Value));
+
+    [Fact]
+    public void Object_General_With_Nested_Property_And_Valid_Input()
```

Comment:
nit: Keeping _with_ to describe the input and avoid use of _and_ (by as much as possible):

```suggestion
    public void Object_General_Nested_With_Valid_Input()
```